frontend/src/scenes/notebooks/Nodes/notebookNodeLogic.ts

```diff
@@ -153,29 +151,42 @@ export const notebookNodeLogic = kea<notebookNodeLogicType>([
 
     listeners(({ actions, values, props }) => ({
         onUpdateEditor: async () => {
+            if (!props.getPos) {
+                return
+            }
             const editor = values.notebookLogic.values.editor
             if (editor) {
-                const pos = props.getPos()
-                const { previous, next } = editor.getAdjacentNodes(pos)
+                const { previous, next } = editor.getAdjacentNodes(props.getPos())
                 actions.setPreviousNode(previous)
                 actions.setNextNode(next)
             }
         },
 
         insertAfter: ({ content }) => {
+            if (!props.getPos) {
```

Comment:
Exactly that. My goal (but maybe not yet) is to abstract the implementation, like you did with the editor, so that it can perform methods that are abstract from the parent engine